solver/src/solver/paraswap_solver.rs

```diff
@@ -57,40 +60,95 @@ impl ParaswapSolver {
     }
 }
 
+pub struct ParaswapCompleteResponse {
+    transaction_result: Result<TransactionBuilderResponse, ParaswapResponseError>,
+    price_response: PriceResponse,
+    amount: U256,
+}
+
 #[async_trait::async_trait]
 impl SingleOrderSolving for ParaswapSolver {
     async fn settle_order(&self, order: LimitOrder) -> Result<Option<Settlement>> {
+        let transaction: TransactionBuilderResponse;
+        let price_response: PriceResponse;
+        let amount: U256;
+        let mut num_retries = 0;
+        loop {
+            let complete_response = self.paraswap_transaction_from_order(&order).await?;
+            if !satisfies_limit_price(&order, &complete_response.price_response) {
+                tracing::debug!("Order limit price not respected");
+                return Ok(None);
+            }
+            match complete_response.transaction_result {
+                Err(ParaswapResponseError::PriceChange) => {
+                    if num_retries >= 2 {
+                        return Err(anyhow!("Price change error after {} retries", num_retries));
+                    }
+                    num_retries += 1;
+                    continue;
+                }
+                _ => {
+                    price_response = complete_response.price_response;
+                    amount = complete_response.amount;
+                    transaction = complete_response.transaction_result?;
+                    break;
```

Comment:
Instead of this you could "return" from the loop:

```rust
let (price_response, amount, transaction) = loop {
...
                    break (
                        complete_response.price_response,
                        complete_response.amount,
                        complete_response.transaction_result?,
                    );
```

Reply:
Hopefully I can also do this in a for loop as was also suggested?